### DISCUSSION TITLE
[community] Kubo not installed

### DISCUSSION THREAD

**Discussion by developersteve** [Lilypad-Tech/lilypad#345] [General]
Using a `Ubuntu 22` instance, attempting to run `ipfs init` results in a kubo not installed error which needs to be installed manually and not via a snap (snap install is not longer supported). 

This originates from the bacalhau error ```bacalhau[8980]: please run: 'ipfs init'```, which then indicates the need to have `kubo` installed via a snap, which will install then return a response about not being supported. 

Additional information https://docs.ipfs.tech/install/command-line/#install-official-binary-distributions

[See the discord thread](https://discord.com/channels/1212897693450641498/1212903910918848542/1238354586880835638)

**Comment by developersteve** [comment]
Manually installing kubo fixes this issue which seems to maybe be only ubuntu specific, its important to note that you cannot use the snap install method that ipfs will recommend. 

@PBillingsby ill get you a VM to test on so you can step through it for doc updates and also to have another set of eyes on it

**Comment by rhochmayr** [comment]
This is how I installed it on Ubuntu 22.04 and 24.04.
https://docs.ipfs.tech/install/command-line/#install-kubo-linux

```
cd ~
wget https://dist.ipfs.tech/kubo/v0.29.0/kubo_v0.29.0_linux-amd64.tar.gz
tar -xvzf kubo_v0.29.0_linux-amd64.tar.gz
cd Kubo/
sudo ./install.sh
```

Now check if ipfs works

```
ipfs --version
```

Afterwards I think the default IPFS path should be set to /app/data/ipfs before running ipfs init.

```
export IPFS_PATH=/app/data/ipfs
cd /app/data/ipfs/
ipfs init
```

**Comment by bgins** [comment]
We did some investigation and this issue is likely due to Bacalhau removing support for embedded IPFS: https://github.com/bacalhau-project/bacalhau/issues/3816. For the moment, the fix is to pin to Bacalhau `v1.3.2` which is the last version with embedded IPFS support. 

We plan to implement another strategy for storing and delivering results (https://github.com/Lilypad-Tech/lilypad/issues/157), and we will announce when pinning to Bacalhau `v1.3.2` is no longer needed.

Closing this issue for now, but please re-open if this wasn't the root cause.

**Comment by rhochmayr** [comment]
This is happening with v1.3.2 though. Didn't the deprecation of embedded IPFS happen in v.1.4.0, which currently isn't compatible with LP anyways?

**Comment by bgins** [comment]
> Didn't the deprecation of embedded IPFS happen in v.1.4.0, which currently isn't compatible with LP anyways?

That's correct. Embedded IPFS has been removed as of `v1.4.0`. Version `v1.3.2` still has it.

> This is happening with v1.3.2 though.

Any chance you have IPFS running somewhere else on your system? Bacalhau might prioritize non-embedded IPFS if it is installed.

@developersteve you got this working after moving to Bacalhau `v1.3.2` right?

**Comment by rhochmayr** [comment]
> Any chance you have IPFS running somewhere else on your system? Bacalhau might prioritize non-embedded IPFS if it is installed.

This was on a clean Ubuntu installation. I'm pretty sure there was no other IPFS instance running or installed. Funnily enough, the same error popped up in the latest AMA video. Around the 34:30 mark. Have a look here. 
https://youtu.be/Q2E65wvbyyU?t=2070

![image](https://github.com/Lilypad-Tech/lilypad/assets/18701385/769529cb-c792-48af-a040-74634c670f83)

The error message instructs the user to run: 'ipfs init'. When you do (on a fresh Ubuntu installation), IPFS is not available and needs to be installed. Unless the embedded version needs to be called in a different way maybe?
